autofs-5.1.8 - coverity fix for invalid access

From: Ian Kent <raven@themaw.net>

Fix invalid access in modules/parse_amd.c:do_host_mount().

Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG           |    1 +
 modules/parse_amd.c |    1 -
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/CHANGELOG b/CHANGELOG
index 279f8de9..c75ff9ad 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -43,6 +43,7 @@
 - fix autofs regression due to positive_timeout.
 - fix parse module instance mutex naming.
 - serialise lookup module open and reinit.
+- coverity fix for invalid access.
 
 19/10/2021 autofs-5.1.8
 - add xdr_exports().
diff --git a/modules/parse_amd.c b/modules/parse_amd.c
index e4909d57..a2684664 100644
--- a/modules/parse_amd.c
+++ b/modules/parse_amd.c
@@ -1373,7 +1373,6 @@ static int do_host_mount(struct autofs_point *ap, const char *name,
 		if (!instance) {
 			error(ap->logopt, MODPREFIX
 			     "failed to create source instance for hosts map");
-			close_lookup(lookup);
 			goto out;
 		}
 	}
